### PR TITLE
Add Github Actions workflow to create a release PR and tag a release

### PR DESCRIPTION
**Pull request by dschuff**
This script runs create_release.py to update the version number and changelog,
and create a tag with the version number on the designated revision.

In order for create_release.py to be runnable without an installation of emscripten
(with a .emscripten_config), the EMSCRIPTEN_REVISION variables were moved from
shared to utils (which does not try to run the config checks at import time).

The action can be run from the web page UI or via the `gh` CLI tool, e.g.
`gh workflow run tag-release.yml -R dschuff/emscripten  -F release-sha=<sha>`

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -569,7 +569,7 @@ def run(args):
     return 0
 
   if '-dumpversion' in args: # gcc's doc states "Print the compiler version [...] and don't do anything else."
-    print(shared.EMSCRIPTEN_VERSION)
+    print(EMSCRIPTEN_VERSION)
```
Comment (sbc100) [inline emcc.py:572]: I think its best to leave this fully qualified since `set_version_globals` won't effect that imported value.

**Review by sbc100 — COMMENTED** [review]
Nice!

I'm being trying to work towards having shared.py be (and all `.py` files actually) be import-able without running any code or requiring a config.   I have an old PR that tried to do this but it fell by the way side.

For now I guess its OK to move these globals, although its not the most natural place for them.   Perhaps you could add a comment "Move these globals back to shared.py once it becomes import-able without a valid config file".

**Comment by sbc100** [comment]
Is the plan to make this automatic?  Doing it via commend line using `gh` or via the web interface by hand doesn't seem much simpler that what we have today.

**Comment by dschuff** [comment]
> Is the plan to make this automatic? Doing it via commend line using `gh` or via the web interface by hand doesn't seem much simpler that what we have today.

That's my hope. The tricky parts are triggering from events that happen outside the repo; for example we want the emsdk workflow to trigger after the LTO build has finished and tested on chromium CI but triggering a workflow requires authentication and the Chromium CI isn't set up to store secret tokens right now. This workflow could run in parallel but we might want it to wait until the emsdk tests pass, so we'd want to trigger it from an action on a separate repo. That seems more likely to be automatable, but I haven't figured out how yet.

I do think that even if we can't go any further than the PRs I've uploaded now, that it will still be an improvement. After landing the Chromium CL, there are now 2 steps rather than 4, and neither of them require a repository to be checked out.

File: tools/utils.py [hunk 2]
```diff
@@ -108,3 +108,11 @@ def delete_contents(dirname, exclude=None):
       delete_dir(entry)
     else:
       delete_file(entry)
+
+
```
Comment (sbc100) [inline tools/utils.py:112]: Can you add a comment:

```
# TODO: Move this back to shared.py once importing that file becoming side effect free (i.e. it no longer requires a config).
```

**Comment by dschuff** [comment]
PTAL, I think this is ready again

**Review by sbc100 — APPROVED** [review]

**Pull request merged**